src/System.Linq/src/System/Linq/Take.cs

```diff
@@ -21,14 +21,12 @@ public static IEnumerable<TSource> Take<TSource>(this IEnumerable<TSource> sourc
                 return EmptyPartition<TSource>.Instance;
             }
 
-            IPartition<TSource> partition = source as IPartition<TSource>;
-            if (partition != null)
+            if (source is IPartition<TSource> partition)
```

Comment:
That seems quite nice (while most of this was inspired by returning to the code after a break and seeing things I'm now used to see done differently, I don't have the same reflexes for pattern `switch` yet). It could be even nicer if I could change which argument is complained about when more than one would trigger an ANE, but that's a very slightly breaking change, so not really worth it.